scripts/test-scripts/compaction-worker-smoke.py

```
@@ -0,0 +1,826 @@
+def _json_request(
+    method: str,
+    url: str,
+    payload: dict | None,
+    auth_key: str,
+    timeout_s: float,
+) -> HttpResult:
+    data = json.dumps(payload).encode("utf-8") if payload is not None else None
+    headers = {"Content-Type": "application/json"}
+    if auth_key:
+        headers["Authorization"] = f"Bearer {auth_key}"
+    req = urllib.request.Request(url, data=data, headers=headers, method=method)
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)